diff --git a/build/index.js b/build/index.js
index 0032244fe0a3adad4c63cd586d00e4a3a28f28bd..24d0d174927baa3ec7cf6c4ae891e8e39c1fe3bd 100644
--- a/build/index.js
+++ b/build/index.js
@@ -79,7 +79,7 @@ function withPigment(nextConfig, pigmentConfig) {
             if (what.startsWith("__barrel_optimize__")) {
               return chunkPOTRHUYT_js.__require.resolve("../next-font");
             }
-            if (what === "next/image" || what === "next/link") {
+            if (what === "next/image" || what === "next/link" || what === "next/link.js") {
               return chunkPOTRHUYT_js.__require.resolve("../next-image");
             }
             if (what.startsWith("next/font")) {
