diff --git a/dist/index.js b/dist/index.js
index 4161f623ba48d04a69ed5358a45a04751f3b9eb1..03b43f76d48b8bd7624a35ea599ce3f6f604bf5e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -173,15 +173,18 @@ async function bundleMDX({
       });
     }
   };
+  /** @type import('esbuild').BuildOptions["define"] */
+  const define = {}
+  if (process.env.NODE_ENV !== undefined) {
+    define['process.env.NODE_ENV'] = JSON.stringify(process.env.NODE_ENV)
+  }
   const buildOptions = esbuildOptions({
     entryPoints: [entryPath],
     write: isWriting,
     outdir: isWriting ? bundleDirectory : undefined,
     publicPath: isWriting ? bundlePath : undefined,
     absWorkingDir: cwd,
-    define: {
-      'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV)
-    },
+    define,
     plugins: [(0, _esbuildPluginGlobalExternals.globalExternals)({
       ...globals,
       react: {